### ISSUE TITLE
Zero-width stroke acts as fill

### ISSUE THREAD

**Issue by dominikh** [linebender/vello#662]
This
```
let stroke = Stroke::new(0.0);
let rect = RoundedRect::new(10.0, 10.0, 240.0, 240.0, 20.0);
let rect_stroke_color = Color::rgb(0.9804, 0.702, 0.5294);
self.scene.stroke(&stroke, Affine::IDENTITY, rect_stroke_color, None, &rect);
```
renders as
![image](https://github.com/user-attachments/assets/89d1fb27-a75e-4c5e-8f2c-6a14ec33fbb3)

**Comment by DJMcNab** [comment]
Thanks for the report! I suspect what's happening here is that the expanded lines overlap, and that confuses something.

But I think @raphlinus will be more likely to give an easy answer.

For the moment, I'd recommend capping your strokes to be at least 0.01 units, presuming that the issue occurs only exactly at 0.

**Comment by dfrg** [comment]
I think the easy fix is to just detect this in `Scene::stroke` and encode nothing. We’d need to do the same for glyph runs.

**Comment by raphlinus** [comment]
Yeah, internally we use a special value of 0.0 of the stroke width to indicate fill. I agree with @dfrg that the easiest thing is to encode nothing. Alternatively, we could switch to using a value of -1.0 (which I believe we did in the past), and have the predicate in the shaders that that the value is negative.

**Comment by TheNachoBIT** [comment]
I've made tests and i found out that every `Style` [(source code just in case)](https://github.com/linebender/vello/blob/main/vello_encoding/src/path.rs) that had a `line_width` of 0.0 gets automatically rendered as a fill no matter what flags in `flags_and_miter_limit` has.

Maybe this is a dumb/not a good idea, but perhaps there could've be a flag for `Style` that explicitly sets if its a fill or a stroke?
